src/relayFeeCalculator/relayFeeCalculator.ts

```diff
@@ -234,26 +237,24 @@ export class RelayFeeCalculator {
       throw new Error(`Could not find token information for ${inputToken}`);
     }
 
-    const outputAmount = getDepositOutputAmount(deposit);
-    const getGasCosts = this.queries
-      .getGasCosts(
-        {
-          ...deposit,
-          amount: simulateZeroFill && isV2Deposit(deposit) ? toBN(100) : outputAmount,
-        },
-        simulateZeroFill ? toBN(100) : amountToRelay,
-        relayerAddress
-      )
-      .catch((error) => {
-        this.logger.error({
-          at: "sdk-v2/gasFeePercent",
-          message: "Error while fetching gas costs",
-          error,
-          simulateZeroFill,
-          deposit,
-        });
-        throw error;
+    if (simulateZeroFill) {
```

Comment:
`simulateZeroFill` is a bit of a misnomer in v3, since we're simulating a reduced fill rather than a `requestV3SlowFill()` call. I've left it as-is because the logic is mostly relevant to the FE, and I think there's plenty of opportunity to revisit that logic once the v2 logic is removed.